p2p/metrics/gossipsub.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	"github.com/spacemeshos/go-spacemesh/metrics"
+)
+
+var (
+	totalPeers       = metrics.NewGauge("total_peers", subsystem, "Total number of peers", nil)
+	peersPerProtocol = metrics.NewGauge("peers_per_protocol", subsystem, "Number of peers per protocol", []string{"protocol"})
+)
+
+// GossipCollector pubsub.RawTracer implementation
+// total number of peers
+// number of peers per each gossip protocol.
+type GossipCollector struct {
+	peers struct {
```

Comment:
maybe it can be removed?

Reply:
no, it need to track when peer is removing. Interface does not provide protocol of removed peer
so we need track this info on our side